code/datums/black_market_item.dm

```diff
@@ -204,18 +196,69 @@ var/locations_calculated = 0
 			A.put_in_any_hand_if_possible(spawned_item)
 
 	log_transaction("The item was teleported directly to him.", user)
-	radio.visible_message("The [radio.name] beeps: <span class='warning'>Thank you for your purchase!</span>")
+	radio.visible_message("The [radio] beeps: <span class='warning'>Thank you for your purchase!</span>")
 	radio.interact(user)
 
-	spawn(rand(30 SECONDS, 60 SECONDS))
-		if(!radio.nanotrasen_variant && prob(sps_chances[EXPENSIVE]))
-			SPS_black_market_alert("Centcomm has detected a black market purchase of item: [name]. It was teleported directly to the buyer in the past minute.")
-
-
 /datum/black_market_item/proc/after_spawn(var/obj/spawned, var/delivery_method, var/mob/user) //Called immediately after spawning. Override for post-spawn behavior.
 	return
 
+/*
+//
+//	PLAYER MARKET DATUM
+//
+*/	
+	
+var/list/player_market_items = list()
+
+/datum/black_market_player_item
+	var/atom/item
+	var/obj/item/device/illegalradio/seller_radio
+	var/mob/living/seller
+	var/selected_name = ""
+	var/selected_price = 100
+	var/selected_description = "Enter description here."
+	
+/datum/black_market_player_item/proc/buy(var/obj/item/device/illegalradio/radio, var/mob/user)
+	..()
+	if(!istype(radio))
+		return FALSE
+	if(user.stat || user.restrained())
+		return FALSE
+	if(!(istype(user,/mob/living/carbon/human)))
+		return FALSE
+	if(!((radio.loc in user.contents) || (in_range(radio.loc, user) && istype(radio.loc.loc, /turf))))
```

Comment:
There has to be a better way than this.

Reply:
Old copy/pasted code from tator uplink, I'm sure I could remove the entirety of the latter half

Reply:
I assume this is to prevent from using a radio within a locked container, but I'm not even sure if this would work as intended